src/DDTrace/Bootstrap.php

```diff
@@ -112,6 +110,12 @@ private static function initRootSpan()
             $span->setTag(Tag::HTTP_URL, $_SERVER['REQUEST_URI']);
             // Status code defaults to 200, will be later on changed when http_response_code will be called
             $span->setTag(Tag::HTTP_STATUS_CODE, 200);
+            // Normalized URL as the resource name
+            $normalizer = new Urls(explode(',', getenv('DD_TRACE_RESOURCE_URI_MAPPING')));
```

Comment:
A couple of comments here, if they make sense to you:

- While I see our conversation about simplifying the way we read configurations, wouldn't you see more strategic to not use `getenv` here? What I mean is that you may just want to move this `getenv()` call to a method in `Config` and not use all the underlying infrastructure.
- I was evaluating pros and cons of doing this here or as a post processing operation. Doing it here means that devs cannot, as an example, set this env in their `index.php` or the likes. In the end we need that info only at the very end, just before we send the trace. Wouldn't you think it is a good idea to push this setting to the very end?

Reply:
Great points!

> What I mean is that you may just want to move this getenv() call to a method in Config and not use all the underlying infrastructure.

I'm certainly open to moving the `getenv()` to an abstraction. :) What do you see as the main advantage for abstracting this one-time env access?

> Wouldn't you think it is a good idea to push this setting to the very end?

This is a great idea! I'll move it to right before the flush... maybe we should start a discussion about possibly adding span filters in a future PR? :)